### PR TITLE
Move Notification Logic from each Layout to their respective Handler

### PR DESCRIPTION
**Pull request by Steffengreiner**
**Purpose of this PR**

This PR moves the notification Logic from the loginlayout and registrationlayout to their respective handler.
Additionally it cleans up the initlayout method by introducing dedicated methods to create and style the individual components.

### CODE REVIEW

**Review by luiskuhn — APPROVED** [review]
I think the refactoring looks appropriate, both Handlers make sense to me

File: webapp/src/main/java/life/qbic/views/login/LoginHandler.java [hunk 1]
```diff
@@ -57,24 +57,36 @@ public void handle(LoginLayout loginView) {
...
   }
 
   private void showEmailConfirmationReminder() {
-    registeredLoginView.showInformation(EMAIL_CONFIRMATION_REMINDER);
+    showInformation(EMAIL_CONFIRMATION_REMINDER);
+  }
+
+  public void clearNotifications() {
+    registeredLoginView.notificationLayout.setVisible(false);
+    registeredLoginView.notificationLayout.removeAll();
```
Comment (jenniferboedker) [inline webapp/src/main/java/life/qbic/views/login/LoginHandler.java:77]: If you remove all you can also leave the layout visible, right?

File: webapp/src/main/java/life/qbic/views/login/LoginHandler.java [hunk 2]
```diff
@@ -57,24 +57,36 @@ public void handle(LoginLayout loginView) {
...
+  }
+
+  public void showError(ErrorMessage errorMessage) {
+    registeredLoginView.notificationLayout.add(
+        new ErrorMessage(errorMessage.title(), errorMessage.message()));
+    registeredLoginView.notificationLayout.setVisible(true);
```
Comment (jenniferboedker) [inline webapp/src/main/java/life/qbic/views/login/LoginHandler.java:83]: why do you create a new message? You could also just forward the error message from the constructor. 
```suggestion
    registeredLoginView.notificationLayout.add(errorMessage);
    registeredLoginView.notificationLayout.setVisible(true);
```

Comment (jenniferboedker) [reply webapp/src/main/java/life/qbic/views/login/LoginHandler.java:83]: If you leave the layout visible you dont need to make it visible again here

File: webapp/src/main/java/life/qbic/views/login/LoginHandler.java [hunk 3]
```diff
@@ -57,24 +57,36 @@ public void handle(LoginLayout loginView) {
...
+  }
+
+  public void showInformation(InformationMessage message) {
+    registeredLoginView.notificationLayout.add(
+        new InformationMessage(message.title(), message.message()));
```
Comment (jenniferboedker) [inline webapp/src/main/java/life/qbic/views/login/LoginHandler.java:88]:
```suggestion
    registeredLoginView.notificationLayout.add(message);
```

File: webapp/src/main/java/life/qbic/views/login/LoginLayout.java [hunk 4]
```diff
@@ -57,18 +53,11 @@ public LoginLayout(@Autowired LoginHandlerInterface loginHandlerInterface) {
 
   private void initLayout() {
     contentLayout = new VerticalLayout();
-    this.loginForm = new ConfigurableLoginForm();
-    loginForm.setAction("login");
```
Comment (jenniferboedker) [inline webapp/src/main/java/life/qbic/views/login/LoginLayout.java:61]: I think we need that. I cant recall but I think it was for the css file. Why did you remove it?

Comment (jenniferboedker) [reply webapp/src/main/java/life/qbic/views/login/LoginLayout.java:61]: Ahh found it nvm!

File: webapp/src/main/java/life/qbic/views/register/UserRegistrationHandler.java [hunk 5]
```diff
@@ -55,17 +65,23 @@ private void addListener() {
...
-    userRegistrationLayout.errorMessage.setVisible(false);
+  private void clearNotifications() {
+    userRegistrationLayout.notificationLayout.setVisible(false);
+    userRegistrationLayout.notificationLayout.removeAll();
+  }
+
+  private void showError(ErrorMessage errorMessage) {
+    userRegistrationLayout.notificationLayout.add(
+        new ErrorMessage(errorMessage.title(), errorMessage.message()));
+    userRegistrationLayout.notificationLayout.setVisible(true);
```
Comment (jenniferboedker) [inline webapp/src/main/java/life/qbic/views/register/UserRegistrationHandler.java:84]: same as above: either directly give the error message and not recreate it make the title and reason as parameters of the method and create the message there

File: webapp/src/main/java/life/qbic/views/login/LoginHandler.java (hunk 2 above)
Comment (Steffengreiner) [reply webapp/src/main/java/life/qbic/views/login/LoginHandler.java:83]: Good point 👍

File: webapp/src/main/java/life/qbic/views/login/LoginHandler.java [hunk 6]
```diff
@@ -98,20 +110,19 @@ public void handle(BeforeEvent beforeEvent) {
...
-    clearMessages();
+    clearNotifications();
     showEmailConfirmationInformation();
-
   }
 
   @Override
   public void onEmailConfirmationFailure(String reason) {
-    registeredLoginView.showError(new ErrorMessage("Email confirmation failed", reason));
+    showError(new ErrorMessage("Email confirmation failed", reason));
```
[the remark this answers is not in the document]
Comment (Steffengreiner) [reply webapp/src/main/java/life/qbic/views/login/LoginHandler.java:124]: However I'd argue that since the errormessage is conveying domain logic I'd keep it being created in the domain logic since the methods intention is to show the error message

[the remark this answers is not in the document]
Comment (jenniferboedker) [reply webapp/src/main/java/life/qbic/views/login/LoginHandler.java:124]: is it really domain logic? its the same class which is part of the infrastucture/application logic

[the remark this answers is not in the document]
Comment (jenniferboedker) [reply webapp/src/main/java/life/qbic/views/login/LoginHandler.java:124]: You get the text from there that is sufficent I think

File: webapp/src/main/java/life/qbic/views/register/UserRegistrationHandler.java (hunk 5 above)
Comment (Steffengreiner) [reply webapp/src/main/java/life/qbic/views/register/UserRegistrationHandler.java:84]: Good point 👍

File: webapp/src/main/java/life/qbic/views/login/LoginHandler.java (hunk 1 above)
Comment (Steffengreiner) [reply webapp/src/main/java/life/qbic/views/login/LoginHandler.java:77]: Since the layout itself does not have margins and paddings i think you're right

File: webapp/src/main/java/life/qbic/views/register/UserRegistrationHandler.java [hunk 7]
```diff
@@ -55,17 +65,23 @@ private void addListener() {
...
               userRegistrationLayout.email.getValue(),
               userRegistrationLayout.password.getValue().toCharArray());
         });
   }
 
-  private void resetErrorMessages() {
-    userRegistrationLayout.alreadyUsedEmailMessage.setVisible(false);
-    userRegistrationLayout.errorMessage.setVisible(false);
+  private void clearNotifications() {
+    userRegistrationLayout.notificationLayout.setVisible(false);
```
[the remark this answers is not in the document]
Comment (Steffengreiner) [reply webapp/src/main/java/life/qbic/views/register/UserRegistrationHandler.java]: Good point

**Review by jenniferboedker — APPROVED** [review]
